packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+        authors
+      `;
+      await expect(
+        async () => await user2TodoHelper.update(updateResultSetName, { id: todo['id'], authors: ['user2'] }, ownersFieldSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // non-owner cannot update a record with an owner protected field
+      const ownerContentFieldSet = `
+        id
+        ownerContent
+      `;
+      await expect(
+        async () => await user2TodoHelper.update(updateResultSetName, { ...todo, ownerContent: 'Owner Content' }, ownerContentFieldSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(updateResultSetName, 'Mutation'));
+
+      // owner cannot read a record with public protected field
+      const getResult1 = await user1TodoHelper.get({ id: todo['id'] }, publicFieldSet, false, 'all');
+      checkOperationResult(
+        getResult1,
+        { id: todo['id'], publicContent: null },
+        `get${modelName}`,
+        false,
+        expectedFieldErrors(['publicContent'], modelName),
+      );
+
+      const listTodosResult1 = await user1TodoHelper.list({}, publicFieldSet, `list${modelName}`, false, 'all');
+      checkListItemExistence(listTodosResult1, `list${modelName}`, todo['id'], true);
+      checkListResponseErrors(listTodosResult1, expectedFieldErrors(['publicContent'], modelName, false));
+
+      /*
+      // non-owner cannot read a record with owner protected fields in the selection set
+      const ownerReadFieldSet = `
+        id
+        owner
+        authors
+        ownerContent
+        ownersContent
+      `;
+      const getResult2 = await user2TodoHelper.get({ id: todo['id'] }, ownerReadFieldSet, false, 'all');
+      checkOperationResult(
+        getResult2, 
+        { id: todo['id'], owner: null, authors: null, ownerContent: null, ownersContent: null }, 
+        `get${modelName}`, 
+        false, 
+        expectedFieldErrors(['owner', 'authors', 'ownerContent', 'ownersContent'], modelName)
+      );
+
+      const listTodosResult2 = await user2TodoHelper.list({}, ownerReadFieldSet, `list${modelName}`, false, 'all');
+      checkListItemExistence(listTodosResult2, `list${modelName}`, todo['id'], true);
+      checkListResponseErrors(listTodosResult2, expectedFieldErrors(['owner', 'authors', 'ownerContent', 'ownersContent'], modelName, false));
+      */
+
+      // non-owner cannot listen to updates on owner protected fields
+      const todoRandom = {
+        ...todo,
+        id: Date.now().toString(),
+      };
+      const todoRandomUpdated = {
+        id: todoRandom.id,
+        owner: userName1,
+        privateContent: 'Private Content updated',
+        ownerContent: 'Owner Content updated',
+      };
+      const subscriberClient = getConfiguredAppsyncClientCognitoAuth(graphQlEndpoint, region, userMap[userName2]);
+      const subTodoHelper = createModelOperationHelpers(subscriberClient, schema)[modelName];
+
+      /*
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket